### PR TITLE
Metalworks E-Regulator Setting

### PR DESCRIPTION
**Pull request by kdumontnu**
logo

**Caution! This will cause any printers running SMC regulators to have poor pressure configuration.**
## Description
This branch involves changes to the regulator.h to correspond with data taken from Metworks Electropneumatic Regulators. This also improves pressure setting in regulator.cpp, so that desired_pressure will not change depending on hysteresis values.

### Requirements
- [X] Unit Test
- [X] Alignment run successfully
- [x] Approval 1 DT
- [X] Approval 2 RR

### CODE REVIEW

**Comment by kdumontnu** [comment]
@pizzyflavin @danthompson41

File: Marlin/Regulator.h [hunk 1]
```diff
@@ -8,14 +8,24 @@
 /*================================================================================*/
 // Calculation for BITS_PER_PSI:
 // R1 = 499
-// R2 (input impedance of sensor) = 10k
+// R2 (input impedance of sensor) = 168k
```
Comment (danthompson41) [inline Marlin/Regulator.h]: Should we mark out that this is Metalworks specific (move it under line 13)?

File: Marlin/Configuration.h [hunk 2]
```diff
@@ -392,8 +392,9 @@ Here are some standard links for getting your machine calibrated:
  // ---------------------------------
  // 0 - No Pressure Sensor Used
  // 1 - Built-in Pressure Transducer (ITV0050-2UL)
+ // 2 - Built-in Pressure Transducer (Regtronics M5)
 
-  #define E_REGULATOR_SENSOR 1
+  #define E_REGULATOR_SENSOR 2
```
Comment (danthompson41) [inline Marlin/Configuration.h]: We might want to consider using defines for 1 and 2 for clarity but with this comment it's not a huge deal

File: Marlin/thermistortables.h [hunk 3]
```diff
@@ -1206,6 +1206,12 @@ const short regulatortable_1[][2] PROGMEM = {
 };
 #endif
 
+#if (E_REGULATOR_SENSOR == 2) // Internal Pressure Sensor LUT for Regtronic M5
+const short regulatortable_2[][2] PROGMEM = {
+  {0*OVERSAMPLENR,       2},
```
Comment (kdumontnu) [inline Marlin/thermistortables.h]: Potentially set to 0 psi

File: Marlin/Regulator.h (hunk 1 above)
Comment (pizzyflavin) [inline Marlin/Regulator.h]: I would leave the original input impedance and label it as SMC or something

File: Marlin/thermistortables.h (hunk 3 above)
Comment (pizzyflavin) [inline Marlin/thermistortables.h]: (to clarify, it currently transforms 0 digital val --> 0.2 psi)

File: Marlin/Regulator.cpp [hunk 4]
```diff
@@ -11,21 +11,26 @@
...
+    // Add half of hysteresis range to desired pressure
+    digital_val = (uint16_t)(BITS_PER_PSI *
+                             (desired_pressure - REG_OFFSET + REG_HYSTERESIS));
+  }
+  // Decreasing pressure
+  else if (desired_pressure < pressureRegulator()) {
+    // Subtract half of hysteresis range from desired pressure
+    digital_val = (uint16_t)(BITS_PER_PSI *
+                             (desired_pressure - REG_OFFSET - REG_HYSTERESIS));
+  }
```
Comment (pizzyflavin) [inline Marlin/Regulator.cpp]: Indentation should be 4 spaces :wine_glass:

File: Marlin/Regulator.h [hunk 5]
```diff
@@ -8,14 +8,24 @@
...
 // n = 12
-// BITS_PER_PSI = (2^n/OUTPUT_PSI_MAX) / (R2/(R1 + R2)) approx = 33.1
```
Comment (pizzyflavin) [inline Marlin/Regulator.h]: Does this formula no longer hold?

File: Marlin/Regulator.h [hunk 6]
```diff
@@ -8,14 +8,24 @@
...
 
-#define BITS_PER_PSI    33.1
+#define BITS_PER_PSI    43.00
 
-#define REG_OFFSET      0.5 // psi
-#define REG_HYSTERESIS  0.2 // psi
+#define REG_OFFSET      0.0  // psi
+#define REG_HYSTERESIS  0.0 // psi
 #define MCP_CONST       (REG_OFFSET - (REG_HYSTERESIS/2.0))
 #define REGULATOR_LOW_P 2
```
Comment (pizzyflavin) [inline Marlin/Regulator.h]: We should probably select the appropriate settings with the preprocessor like

``` c
#if E_REGULATOR == 1
    #define OFFSET blabllah
    #define HYST blah
#elif E_REGULATOR == 2
    #define OFFSET blahblah
    #define HYST blahlll
#endif
```

**Comment by pizzyflavin** [comment]
@kdumontnu Hit us up when changes are made or addressed so we can approve!

File: Marlin/Regulator.cpp (hunk 4 above)
Comment (kdumontnu) [inline Marlin/Regulator.cpp]: But should they?? O.o
I'll fight this one another time. Touché.

File: Marlin/thermistortables.h (hunk 3 above)
Comment (kdumontnu) [inline Marlin/thermistortables.h]: Looking at the the data, it is correct as is. No change.

**Comment by kdumontnu** [comment]
@pizzyflavin good to go

**Comment by pizzyflavin** [comment]
After that comment is fixed and the hysteresis is divided by two in the #define, it LGTM

**Pull request merged**